packages/yourloops/test/unit/pages/stepper/signup-profile-form.test.tsx

```diff
@@ -27,65 +27,47 @@
  */
 
 import React from 'react'
-import { render, unmountComponentAtNode } from 'react-dom'
-import { act } from '@testing-library/react-hooks/dom'
-import _ from 'lodash'
 
-import { SignUpFormStateProvider } from '../../../../pages/signup/signup-formstate-context'
 import SignUpProfileForm from '../../../../pages/signup/signup-profile-form'
+import { SignupForm } from '../../../../lib/auth'
 import * as authHookMock from '../../../../lib/auth'
-import User from '../../../../lib/auth/user'
+import * as signupFormProviderMock from '../../../../pages/signup/signup-formstate-context'
+
 import { UserRoles } from '../../../../models/user'
+import { render, screen } from '@testing-library/react'
 
 jest.mock('../../../../lib/auth')
+jest.mock('../../../../pages/signup/signup-formstate-context')
 describe('Signup profile form', () => {
-  let container: HTMLElement | null = null
-
-  const mountComponent = (): void => {
-    act(() => {
-      render(
-        <SignUpFormStateProvider>
-          <SignUpProfileForm handleBack={_.noop} handleNext={_.noop} />
-        </SignUpFormStateProvider>, container)
-    })
+  const mountComponent = () => {
+    return render(
+      <SignUpProfileForm handleBack={jest.fn} handleNext={jest.fn} />
+    )
   }
 
-  beforeEach(() => {
-    container = document.createElement('div')
-    document.body.appendChild(container);
-    (authHookMock.useAuth as jest.Mock).mockImplementation(() => {
-      return {
-        user: {
-          role: UserRoles.hcp
-        } as User
-      }
-    })
-  })
-
-  afterEach(() => {
-    if (container) {
-      unmountComponentAtNode(container)
-      container.remove()
-      container = null
-    }
+  beforeAll(() => {
+    (authHookMock.useAuth as jest.Mock).mockImplementation(() => ({
+      completeSignup: jest.fn()
+    }));
+    (signupFormProviderMock.useSignUpFormState as jest.Mock).mockImplementation(() => ({
+      state: { accountRole: UserRoles.caregiver }
+    }))
   })
 
   it('should not render the drop down list when caregiver', () => {
-    (authHookMock.useAuth as jest.Mock).mockImplementation(() => {
-      return {
-        user: {
-          role: UserRoles.caregiver
-        } as User
-      }
-    })
     mountComponent()
-    const dropDownList = document.querySelector('#hcp-profession-selector')
-    expect(dropDownList).toBeNull()
+    expect(screen.queryByTestId('hcp-profession-selector')).not.toBeInTheDocument()
   })
 
   it('should render the drop down list when HCP', () => {
+    (signupFormProviderMock.useSignUpFormState as jest.Mock).mockImplementation(() => {
```

Comment:
If this test is checked via one of your integration test, then you can delete it :slightly_smiling_face: